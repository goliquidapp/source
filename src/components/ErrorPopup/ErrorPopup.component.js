import React, {Component} from 'react';
import {Text, View} from 'react-native';
import {Icon} from 'react-native-elements'

import Colors from '../../resources/Colors.js';
import TextTicker from 'react-native-text-ticker'

class ErrorPopup extends Component{
	constructor(props){
		super(props);
		this.state={visible:true}
		this.timeout=100;
		this.timeoutObj=null;
	}
	componentDidMount(){
		this.toggle()
	}
	componentDidUpdate(prevProps){
		if (prevProps.message!==this.props.message){
			this.setState({visible:true},this.toggle)
		}
	}
	componentWillUnmount(){
		clearTimeout(this.timeoutObj)
	}
	toggle=()=>{
		this.timeoutObj=setTimeout(()=>this.setState({visible:false}),this.timeout*this.props.message.length)
	}
	render(){
		const {containerStyle,messageStyle,message,visible}=this.props
		if (this.state.visible){
			return (
				<View style={[styles.container,containerStyle]}>
					<Icon name="error" type='material-icons' size={20} color={Colors['White']}/>
					<TextTicker
                        repeatSpacer={50}
                        marqueeDelay={this.timeout*4}
                        duration={this.timeout*this.props.message.length}
                        style={[styles.message,messageStyle]}>

                        {message}

                    </TextTicker>
				</View>
			)
		}else{
			return <View></View>
		}
	}
}

const styles={
	container:{
		flexDirection:'row',
		alignItems:'center',
		justifyContent:'center',
		width:'100%',
		height:50,
		position:'absolute',
		backgroundColor:'rgba(255,0,0,0.8)',
		bottom:0,
		left:0,
		zIndex:100,
		elevation:8,
		paddingHorizontal:10
	},
	message:{
		color:Colors['White'],
		textAlign:'center',
		paddingHorizontal:10
	}
}

export default ErrorPopup