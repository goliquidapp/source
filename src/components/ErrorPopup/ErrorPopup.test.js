import React from 'react';
import { Text, View} from 'react-native';
import renderer from 'react-test-renderer';

import ErrorPopup from './ErrorPopup.component';

describe("Testing ErrorPopup component",()=>{
	it('renders correctly', () => {
		const wrapper = renderer.create(
    		<ErrorPopup message={"Something wrong!"}/>
    	).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
})