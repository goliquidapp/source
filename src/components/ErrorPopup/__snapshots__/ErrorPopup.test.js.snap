// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ErrorPopup component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255,0,0,0.8)",
        "bottom": 0,
        "elevation": 8,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "left": 0,
        "paddingHorizontal": 10,
        "position": "absolute",
        "width": "100%",
        "zIndex": 100,
      },
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
      underlayColor="white"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        î€€
      </Text>
    </View>
  </View>
  <Text>
    Something wrong!
  </Text>
</View>
`;
